src/librustc_yk_sections/mir_cfg.rs

```diff
@@ -64,43 +150,192 @@ pub fn generate_tir<'a, 'tcx, 'gcx>(
     Ok(ret)
 }
 
+/// Lazy computation of dominance frontiers.
+///
+/// See Page 404 of the 2nd edition of the Appel book mention above.
+///
+/// Since the frontier of one node may depend on the frontiers of other nodes (depending upon their
+/// relationships), we cache the frontiers so as to avoid computing things more than once.
+struct DominatorFrontiers<'a, 'tcx> {
+    /// The MIR we are working on.
+    mir: &'a Mir<'tcx>,
+    /// The dominators of the above MIR.
+    doms: Dominators<BasicBlock>,
+    /// The dominance frontiers. Lazily computed. None means as-yet uncomputed.
+    df: IndexVec<BasicBlock, Option<Vec<BasicBlock>>>,
+}
+
+impl<'a, 'tcx> DominatorFrontiers<'a, 'tcx> {
+    fn new(mir: &'a Mir<'tcx>) -> Self {
+        let num_blks = mir.basic_blocks().len();
+        let mut df = IndexVec::with_capacity(num_blks);
```

Comment:
Does `IndexVec` have a `from_elem` function? If so, it'll probably execute faster than this + the for loop.

Reply:
Yes, it does. Much neater.